docker/start_build.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -e
+
+LOG_DIR="${HOME}/docker/logs"
+GIT_LOG="${LOG_DIR}/git.log"
+BUILD_LOG="${LOG_DIR}/build.log"
+
+cd ~/
+
+[ -n "${TAG}" ] || TAG="master"; \
+{ git pull; [ ${TAG} = 'master' ] || { git describe; git verify-tag ${TAG}; }; \
+git verify-commit ${TAG}^{commit}; git checkout --recurse-submodules ${TAG}; \
+git status; } 2>&1 | tee -a ${GIT_LOG}
```

Comment:
This is very hard to read and should be expanded some, perhaps like this:

```suggestion
{
  if [ -z "${TAG:-}" ]; then
    TAG="master";
  fi
  git pull
  if [ "${TAG}" != 'master' ]; then
    git describe
    git verify-tag "${TAG}"
  fi
  git verify-commit "${TAG}^{commit}"
  git checkout --recurse-submodules "${TAG}"
  git status
} 2>&1 | tee -a -- "${GIT_LOG}"
```